File: server/models/Order.js
import mongoose from "mongoose";

const productSchema = new mongoose.Schema({
  productId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: "Product",
  },
  name: {
    type: String,
    required: true,
  },
  price: {
    type: Number,
    required: true,
  },
  size: {
    type: String,
  },
  imageUrl: {
    type: String,
  },
});

const orderSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: "User",
  },
  name: {
    type: String,
    required: true,
  },

  status: { type: Number, default: 0 },
  txnRef: { type: String },
  email: {
    type: String,
    required: true,
  },
  phone: {
    type: String,
    required: true,
  },
  address: {
    type: String,
    required: true,
  },
  note: {
    type: String,
  },
  orderTime: {
    type: Date,
    required: true,
  },
  totalAmount: {
    type: Number,
    required: true,
  },
  paymentMethod: {
    type: String,
    required: true,
  },
  products: [productSchema],
});

const Order = mongoose.model("Order", orderSchema);

export default Order;

File: backend-dotnet/ECommerce.Api/Entities/Order.cs
namespace ECommerce.Api.Entities;

public class Order : BaseEntity
{
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Status { get; set; } = 0; // 0=Pending, 1=Confirmed, 2=Shipping, 3=Delivered, -2=Failed
    public string? TxnRef { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string? Note { get; set; }
    public DateTime OrderTime { get; set; }
    public decimal TotalAmount { get; set; }
    public string PaymentMethod { get; set; } = string.Empty;

    // Navigation
    public User User { get; set; } = null!;
    public ICollection<OrderItem> Products { get; set; } = new List<OrderItem>();
}
